test/Microsoft.ComponentDetection.Detectors.Tests/RustCliDetectorTests.cs

```diff
@@ -292,4 +293,266 @@ public async Task RustCliDetector_InvalidNameAsync()
+      ""metadata"": {},
+      ""publish"": null,
+      ""authors"": [
+        ""Sample Author 1"",
+        ""Sample Author 2""
+      ],
+      ""categories"": [],
+      ""keywords"": [],
+      ""readme"": ""README.md"",
+      ""repository"": ""https://github.com/tkaitchuck/ahash"",
+      ""homepage"": null,
+      ""documentation"": """",
+      ""edition"": ""00"",
+      ""links"": null,
+      ""default_run"": null,
+      ""rust_version"": null
+    },
+    {
+      ""name"": ""rust-test"",
+      ""version"": ""0.1.0"",
+      ""id"": ""rust-test (registry+https://github.com/rust-lang/crates.io-index)"",
+      ""license"": ""MIT"",
+      ""license_file"": null,
+      ""description"": ""A non-cryptographic hash function using AES-NI for high performance"",
+      ""source"": ""registry+https://github.com/rust-lang/crates.io-index"",
+      ""dependencies"": [],
+      ""targets"": [],
+      ""features"": {},
+      ""manifest_path"": """",
+      ""metadata"": {},
+      ""publish"": null,
+      ""authors"": [
+        ""Sample Author 1"",
+        ""Sample Author 2""
+      ],
+      ""categories"": [],
+      ""keywords"": [],
+      ""readme"": ""README.md"",
+      ""repository"": ""https://github.com/tkaitchuck/ahash"",
+      ""homepage"": null,
+      ""documentation"": """",
+      ""edition"": ""000"",
+      ""links"": null,
+      ""default_run"": null,
+      ""rust_version"": null
+    }
+],
+    ""target_directory"": ""/home/justin/rust-test/target"",
+    ""version"": 1,
+    ""workspace_root"": ""/home/justin/rust-test"",
+    ""metadata"": null
+}";
+        this.mockCliService.Setup(x => x.CanCommandBeLocatedAsync("cargo", It.IsAny<IEnumerable<string>>())).ReturnsAsync(true);
+        this.mockCliService.Setup(x => x.ExecuteCommandAsync(It.IsAny<string>(), It.IsAny<IEnumerable<string>>(), It.IsAny<string[]>()))
+            .ReturnsAsync(new CommandLineExecutionResult { StdOut = cargoMetadata });
+
+        var (scanResult, componentRecorder) = await this.DetectorTestUtility
+            .WithFile("Cargo.toml", string.Empty)
+            .ExecuteDetectorAsync();
+
+        scanResult.ResultCode.Should().Be(ProcessingResultCode.Success);
+        componentRecorder.GetDetectedComponents().Should().HaveCount(2);
+
+        componentRecorder
+            .GetDetectedComponents()
+            .Select(x => x.Component.Id)
+            .Should()
+            .BeEquivalentTo("libc 0.2.147 - Cargo", "rust-test 0.1.0 - Cargo");
+
+        var components = componentRecorder.GetDetectedComponents();
+
+        foreach (var component in components)
+        {
+            if (component.Component is CargoComponent cargoComponent)
+            {
+                cargoComponent.Author.Should().Be("Sample Author 1, Sample Author 2");
+                cargoComponent.License.Should().Be("MIT");
+            }
+        }
+    }
+
+    [TestMethod]
+    public async Task RustCliDetector_AuthorAndLicenseNullAsync()
```

Comment:
Can you add a similar test for empty string values to validate the `x.Authors.All(a => string.IsNullOrWhiteSpace(a))` logic?

Reply:
Done